include/enumerations/mesh_entities.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+namespace specfem::mesh_entity {
+
+enum class type : int {
+  bottom = 1,
+  top = 2,
+  left = 3,
+  right = 4,
+  bottom_left = 5,
+  bottom_right = 6,
+  top_right = 7,
+  top_left = 8
+};
```

Comment:
Is there a significance to this ordering? It may not be worth changing from what we already have, but I would think it to make more sense if these matched the `QUA_9` ([`NGNOD = 9`](https://specfem2d.readthedocs.io/en/latest/03_mesh_generation/#notes-about-datapar_file-parameters)) node order:
```
         4 . . . . 7 . . . . 3
         .                   .
         .         eta       .
         .         |         .
         8         9--xi     6
         .                   .
         .                   .
         .                   .
         1 . . . . 5 . . . . 2
```
that is,
```suggestion
enum class type : int {
  bottom = 5,
  top = 7,
  left = 8,
  right = 6,
  bottom_left = 1,
  bottom_right = 2,
  top_right = 3,
  top_left = 4
};
```

Reply:
It's historical, the fortan mesher uses this numbering when writing interfaces, boundaries. Changing here would require changing the mesher and then we'd deviate from original mesher/solver conventions